src/plugin/fieldViews/RepeaterFieldView.ts

```diff
@@ -85,28 +115,100 @@ export class RepeaterFieldView extends FieldView<unknown> {
       );
       return;
     }
-    tr.replaceWith(endOfRepeaterNode, endOfRepeaterNode, newNode);
+    let positionToAddFrom;
+    if (maybeIndex === undefined) {
+      // If no index, add to end of repeater node
+      positionToAddFrom = this.getPos() + this.offset + this.node.nodeSize;
+    } else {
```

Comment:
We still add child nodes at the end of the repeater when no index is supplied - we could deprecate this function.

Reply:
@fredex42 is 'adding at the end' something that's going to be needed by recipes use case... if not we should remove to avoid premature abstraction

Reply:
@twrichards I am OK to remove this.... I figure that "insert" is a better case for us as it means less scrolling in the UI